airbyte-integrations/connectors/destination-redshift/src/test-integration/java/io/airbyte/integrations/destination/redshift/RedshiftStagingS3DestinationAcceptanceTest.java

```diff
@@ -203,8 +219,35 @@ private List<JsonNode> retrieveRecordsFromTable(final String tableName, final St
         ctx -> ctx
             .fetch(String.format("SELECT * FROM %s.%s ORDER BY %s ASC;", schemaName, tableName, JavaBaseConstants.COLUMN_NAME_EMITTED_AT))
             .stream()
-            .map(this::getJsonFromRecord)
-            .collect(Collectors.toList()));
+            .map(record -> getJsonFromRecord(
+                record,
+                value -> {
+                  if (value instanceof final RedshiftTimestamp rts) {
+                    // We can't just use rts.toInstant().toString(), because that will mangle historical
+                    // dates (e.g. 1504-02-28...) because toInstant() just converts to epoch millis,
+                    // which works _very badly_ for for very old dates.
```

Comment:
I guess anything before Epoch (1970 ?) will be broken for `toInstant()`

Reply:
iirc there's some really dumb calendar changes (i.e dates being skipped) around the 1600s. same as how snowflake [recommends](https://docs.snowflake.com/en/sql-reference/data-types-datetime#date) avoiding using dates prior to 1582

... which raises the question of whether we even want to support those dates, tbh